README.md

```diff
@@ -5,757 +5,864 @@
+
+let meaningOfLifeAndEverything = create(Number, 42);
+meaningOfLifeAndEverything.state
+//> 42
+
+let greeting = create(String, 'Hello World');
+greeting.state
+//> Hello World
+
+let isOpen = create(Boolean, true);
+isOpen.state
+//> true
 
-create(Counter);
-// => {
-//      merge: Function
-//      set: Function
-//      count: {
-//        increment: Function
-//        decrement: Function
-//        sum: Function
-//        subtract: Function
-//      }
-//    }
+let foo = create(Object, { foo: 'bar' });
+foo.state
+//> { foo: 'bar' }
 
-Microstate.create(Counter).state;
-// => { count: 0 }
+let numbers = create(Array, [ 1, 2, 3, 4 ]);
+numbers.state
+//> [ 1, 2, 3, 4 ]
 ```
 
-We use Class Properties syntax to define the composition of microstates. If you're using Babel, you need the
-[Class Properties transform](https://babeljs.io/docs/plugins/transform-class-properties/) for Class
-Properties to work.
+Apart from these basic types, every other type in Microstates is composed out of other types. So for example, to create a `Person` you would define a JavaScript class and declare composed types with class properties(aka class fields).
 
 ```js
 class Person {
   name = String;
   age = Number;
 }
+```
 
-create(Person);
-// => {
-//      name: { concat: Function, set: Function },
-//      age: { increment: Function, decrement: Function, sum: Function, subtract: Function }
-//    }
+Once you have a type, you can use that type to create as many people as your application requires:
 
-// state objects maintain their type's class
-create(Person).state instanceof Person;
-// => true
+```js
+import { create } from "microstates";
 
-create(Person).state;
-// => { name: '', age: 0 }
+let person = create(Person, { name: "Homer", age: 39 });
 ```
 
-You can restore a composed microstate by providing initial value that matches the structure of the
-microstate.
+Every microstate created with a type of `Person` will be an object that looks like this:
 
-```js
-create(Person, { name: 'Taras', age: 99 }).state;
-// => { name: 'Taras', age: 99 }
+```txt
++----------------------+
+|                      |       +--------------------+
+|  Microstate<Person>  +-name--+                    +-concat()->
+|                      |       | Microstate<String> +-set()->
+|                      |       |                    +-state: 'Homer'
+|                      |       +--------------------+
+|                      |
+|                      |       +--------------------+
+|                      +-age---+                    +-increment()->
+|                      |       | Microstate<Number> +-decrement()->
+|                      |       |                    +-set()->
+|                      |       |                    +-state: 39
+|                      |       +--------------------+
+|                      |
+|                      +-set()->
+|                      +-state: Person { name: 'Homer', age: 39 }
++----------------------+
 ```
 
```

Comment:
I think it's worth calling out that the number microstate has transitions and that we'll see later how to invoke them, but worth introducing the idea that we don't have to write any.